h/services/rename_user.py

```diff
@@ -57,8 +56,13 @@ def rename(self, user, new_username):
         # can just update the userid.
         self._update_tokens(old_userid, new_userid)
 
-        ids = self._change_annotations(old_userid, new_userid)
-        self.reindex(ids)
+        self._change_annotations(old_userid, new_userid)
+        self._search_index.add_users_annotations(
+            old_userid,
+            tag="RenameUserService.rename()",
+            force=True,
+            schedule_in=30,
+        )
```

Comment:
As with NIPSA we're changing the DB and then reindexing, so we need to make sure the reindexing doesn't happen until after the request's transaction commits, so we use a 30 sec delay.